sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/executable/SonarTSCoreBundle.java

```diff
@@ -86,38 +82,21 @@ private void deploy() {
   }
 
   /**
-   * Builds command to run tslint
+   * Builds command to run rules with tsrunner
    */
   @Override
-  public Command getTslintCommand(String tsconfigPath, Collection<InputFile> inputFiles) {
-    File sonartsCoreDir = new File(deployDestination, "sonarts-core");
-
-    Command command = Command.create("node");
-    command.addArgument(tslintExecutable.getAbsolutePath());
-    command.addArgument("--config").addArgument(new File(sonartsCoreDir, "tslint.json").getAbsolutePath());
-    command.addArgument("--format").addArgument("json");
-    // "--force" parameter will force execution of the rules even in case of compilation error
-    command.addArgument("--force");
-
-    command.addArgument("--type-check")
-      .addArgument("--project")
-      .addArgument(tsconfigPath);
-
-    for (InputFile inputFile : inputFiles) {
-      command.addArgument(inputFile.absolutePath());
-    }
-
-    return command;
+  public SonarTSRunnerCommand getRuleRunnerCommand(String tsconfigPath, Collection<InputFile> inputFiles) {
+    SonarTSRunnerCommand runnerCommand = new SonarTSRunnerCommand("node", this.tsMetricsExecutable.getAbsolutePath());
+    return runnerCommand;
   }
 
   /**
    * Builds command to run "sonar", which is making side information calculation (metrics, highlighting etc.)
    */
   @Override
-  public Command getTsMetricsCommand() {
-    Command command = Command.create("node");
-    command.addArgument(this.tsMetricsExecutable.getAbsolutePath());
-    return command;
+  public SonarTSRunnerCommand getTsMetricsCommand() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "getRuleRunnerCommand" on line 88. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4144)